iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/evaluator/CompactionEvaluator.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.iceberg.mr.hive.compaction.evaluator.amoro.TableFileScanHelper;
+import org.apache.iceberg.mr.hive.compaction.evaluator.amoro.TableFormat;
+import org.apache.iceberg.mr.hive.compaction.evaluator.amoro.TableRuntime;
+import org.apache.iceberg.mr.hive.compaction.evaluator.amoro.TableRuntimeMeta;
+import org.apache.iceberg.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CompactionEvaluator {
+
+  private CompactionEvaluator() {
+
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompactionEvaluator.class);
+
+  public static boolean isEligibleForCompaction(Table icebergTable, String partitionPath,
+      CompactionType compactionType, HiveConf conf) {
+
+    if (icebergTable.currentSnapshot() == null) {
+      LOG.info("Table {}{} doesn't require compaction because it is empty", icebergTable,
+          partitionPath == null ? "" : " partition " + partitionPath);
+      return false;
+    }
+
+    CommonPartitionEvaluator partitionEvaluator = constructCommonPartitionEvaluator(icebergTable, partitionPath, conf);
+
+    switch (compactionType) {
+      case MINOR:
+        return partitionEvaluator.isMinorNecessary();
+      case MAJOR:
+        return partitionEvaluator.isFullNecessary() || partitionEvaluator.isMajorNecessary();
+      default:
+        return false;
+    }
+  }
+
+  private static TableRuntime constructTableRuntime(Table icebergTable, HiveConf conf) {
+    long fileSizeInBytesThreshold = HiveConf.getSizeVar(conf,
+        HiveConf.ConfVars.HIVE_ICEBERG_COMPACTION_FILE_SIZE_THRESHOLD);
+
+    OptimizingConfig optimizingConfig = OptimizingConfig.parse(Collections.emptyMap());
+    optimizingConfig.setTargetSize(fileSizeInBytesThreshold);
+    optimizingConfig.setFullTriggerInterval(0);
+    optimizingConfig.setMinorLeastInterval(0);
+
+    TableConfiguration tableConfig = new TableConfiguration();
+    tableConfig.setOptimizingConfig(optimizingConfig);
+
+    TableRuntimeMeta tableRuntimeMeta = new TableRuntimeMeta();
+    tableRuntimeMeta.setTableName(icebergTable.name());
+    tableRuntimeMeta.setFormat(TableFormat.ICEBERG);
+    tableRuntimeMeta.setLastFullOptimizingTime(1);
+    tableRuntimeMeta.setLastMinorOptimizingTime(1);
+    tableRuntimeMeta.setTableConfig(tableConfig);
+
+    return new HiveTableRuntime(tableRuntimeMeta);
+  }
+
+  private static CommonPartitionEvaluator constructCommonPartitionEvaluator(Table table, String partitionPath,
+      HiveConf conf) {
+    TableRuntime tableRuntime = constructTableRuntime(table, conf);
+
+    TableFileScanHelper tableFileScanHelper = new IcebergTableFileScanHelper(table,
+        table.currentSnapshot().snapshotId());
+    CommonPartitionEvaluator evaluator = null;
+    try (CloseableIterable<TableFileScanHelper.FileScanResult> results =
+             tableFileScanHelper.scan()) {
+      for (TableFileScanHelper.FileScanResult fileScanResult : results) {
+        DataFile file = fileScanResult.file();
```

Comment:
would it be extended with `deletes` as well?

Reply:
no, it's not needed.
I use `file` only to get `partitionSpec` and to check if this file and its deletes need to be handled when processing the given partition.
If the file does need to be handled with the given partitionPath, I'm adding the data file and the delete files to the compaction evaluator:
`evaluator.addFile(fileScanResult.file(), fileScanResult.deleteFiles());`